### ISSUE TITLE
2.5.4 shows footer link colour as color: #2c2d33 irregardless of value chosen in Customizer

### ISSUE THREAD

**Issue by dougaitken** [woocommerce/storefront#1269]
18826995-hc

## Describe the bug

Upgrading Storefront to `2.5.4` makes any footer link color #2c2d33 instead of the color chosen in the **Customizer > Footer** setting

Isolating the problem (mark completed items with an [x]):
- [x] I have deactivated other plugins and themes and confirmed this bug occurs when only WooCommerce + Storefront theme are active.
- [x] I can reproduce this bug consistently using the steps below.

## To Reproduce
1. Go to **Customizer > Footer** 
2. Set Link color as something
3. Upgrade Storefront to 2.5.4
4. See color is now set to #2c2d33

## Screenshots

Screencast: https://d.pr/i/bp4HSc

## Expected behavior
Changing any Customizer settings has a direct effect on the site.

### Browser Environment
Please provide as much detail as possible about your testing environment.

- Platform: e.g. macOS
- Browser(s): latest Firefox

### WordPress Environment
Please provide relevant details of your WordPress setup and server environment.

<details>
```
`
### WordPress Environment ###

WordPress address (URL): https://core.doug.press
Site address (URL): https://core.doug.press
WC Version: 3.9.2
REST API Version: ✔ 1.0.7
Log Directory Writable: ✔
WP Version: 5.3.2
WP Multisite: –
WP Memory Limit: 256 MB
WP Debug Mode: –
WP Cron: ✔
Language: en_US
External object cache: ✔

### Server Environment ###

Server Info: nginx
PHP Version: 7.2.27
PHP Post Max Size: 2 GB
PHP Time Limit: 1200
PHP Max Input Vars: 6144
cURL Version: 7.68.0
OpenSSL/1.1.0l

SUHOSIN Installed: –
MySQL Version: 5.5.5-10.3.21-MariaDB-log
Max Upload Size: 2 GB
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ✔
DOMDocument: ✔
GZip: ✔
Multibyte String: ✔
Remote Post: ✔
Remote Get: ✔

### Database ###

WC Database Version: 3.9.2
WC Database Prefix: wp_
Total Database Size: 2.82MB
Database Data Size: 1.66MB
Database Index Size: 1.16MB
wp_woocommerce_sessions: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_api_keys: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_attribute_taxonomies: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_downloadable_product_permissions: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_woocommerce_order_items: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_order_itemmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_tax_rates: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_woocommerce_tax_rate_locations: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_shipping_zones: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_woocommerce_shipping_zone_locations: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_shipping_zone_methods: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_woocommerce_payment_tokens: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_payment_tokenmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_log: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_advanced_notifications: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_advanced_notification_triggers: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_commentmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_comments: Data: 0.02MB + Index: 0.09MB + Engine InnoDB
wp_links: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_options: Data: 0.31MB + Index: 0.06MB + Engine InnoDB
wp_postmeta: Data: 0.28MB + Index: 0.16MB + Engine InnoDB
wp_posts: Data: 0.34MB + Index: 0.06MB + Engine InnoDB
wp_termmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_terms: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_term_relationships: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_term_taxonomy: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_usermeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_users: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_wc_deposits_payment_plans: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wc_deposits_payment_plans_schedule: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_download_log: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_product_meta_lookup: Data: 0.02MB + Index: 0.09MB + Engine InnoDB
wp_wc_tax_rate_classes: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_webhooks: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wpml_mails: Data: 0.11MB + Index: 0.00MB + Engine InnoDB

### Post Type Counts ###

attachment: 49
custom_css: 1
customize_changeset: 12
global_product_addon: 1
jetpack_migration: 2
jp_img_sitemap: 1
jp_sitemap: 1
jp_sitemap_master: 1
log_emails_log: 15
nav_menu_item: 8
page: 12
post: 3
product: 29
product_variation: 12
revision: 35
scheduled-action: 5
shop_coupon: 2
shop_order: 12

### Security ###

Secure connection (HTTPS): ✔
Hide errors from visitors: ✔

### Active Plugins (3) ###

Akismet Anti-Spam: by Automattic – 4.1.3
Log Emails: by WebAware – 1.3.1
WooCommerce: by Automattic – 3.9.2

### Inactive Plugins (11) ###

Classic Editor: by WordPress Contributors – 1.5
Gutenberg: by Gutenberg Team – 7.3.0
Homepage Control: by WooThemes – 2.0.3
Jetpack by WordPress.com: by Automattic – 8.2.1
Storefront Parallax Hero: by WooCommerce – 1.5.7
WC Minimum Order Amount: by Hannah Swain – 1.0 – Not tested with the active version of WooCommerce
WooCommerce Blocks: by Automattic – 2.5.13
WooCommerce One Page Checkout: by Automattic – 1.7.3
WooCommerce Slack: by WooCommerce – 1.2.0
WooCommerce Stripe Gateway: by WooCommerce – 4.3.2
WP Mail Logging: by MailPoet – 1.9.5

### Dropin Plugins (2) ###

advanced-cache.php: advanced-cache.php
object-cache.php: Memcached

### Settings ###

API Enabled: –
Force SSL: –
Currency: GBP (£)
Currency Position: left
Thousand Separator: ,
Decimal Separator: .
Number of Decimals: 2
Taxonomies: Product Types: external (external)
grouped (grouped)
simple (simple)
variable (variable)

Taxonomies: Product Visibility: exclude-from-catalog (exclude-from-catalog)
exclude-from-search (exclude-from-search)
featured (featured)
outofstock (outofstock)
rated-1 (rated-1)
rated-2 (rated-2)
rated-3 (rated-3)
rated-4 (rated-4)
rated-5 (rated-5)

Connected to WooCommerce.com: ✔

### WC Pages ###

Shop base: #5 - /shop/
Cart: #6 - /cart/
Checkout: #7 - /checkout/
My account: #8 - /my-account/
Terms and conditions: ❌ Page not set

### Theme ###

Name: Storefront
Version: 2.5.4
Author URL: https://woocommerce.com/
Child Theme: ❌ – If you are modifying WooCommerce on a parent theme that you did not build personally we recommend using a child theme. See: How to create a child theme
WooCommerce Support: ✔

### Templates ###

Overrides: –

### Action Scheduler ###

Complete: 5
Oldest: 2020-01-22 10:56:56 +0000
Newest: 2020-02-13 14:23:43 +0000

Pending: 0
Oldest: –
Newest: –

Canceled: 0
Oldest: –
Newest: –

In-progress: 0
Oldest: –
Newest: –

Failed: 0
Oldest: –
Newest: –

`
```
</details>

**Comment by billrobbins** [comment]
This style is more specific which is preventing the customizer's color from working:

`
.site-footer a:not(.button):not(.components-button) {
    color: #2c2d33;
    text-decoration: underline;
}
`

Here's the output from the customizer setting for reference:

`
.site-footer a:not(.button) {
     color: #8224e3;
}
`

**Comment by haszari** [comment]
Thanks for the report @dougaitken & workaround @billrobbins. This issue was also reported in #1260 - and the fix has now been merged 🎉  (targeting 2.5.5).

I'm closing this duplicate report – please reopen if this is a different issue that needs another fix.
